File: examples/agent-quickstart/quickstart.mjs
// Prism agent quickstart: authenticate with a wallet, lease a GPU, run a command,
// and read the metered result — no browser, no dashboard.
//
//   PRISM_AGENT_KEY=0x<agent wallet private key> \
//   PRISM_ESCROW=0x71Df0eF3bc81022cB3bec0b1a05f52f12bAfcDeD \
//   node quickstart.mjs
//
// The wallet needs USDG and native Robinhood-Chain gas. Set PRISM_RUN_LEASE=1 to
// actually lease and run (spends funds); otherwise this stops after listing GPUs.
// Prism is pre-production and unaudited — do not use funds or data you cannot lose.
import { DEFAULT_IMAGE, PrismAgent } from "@prism-network/agent-sdk";

const agent = new PrismAgent({
  privateKey: requireEnv("PRISM_AGENT_KEY"),
  escrow: requireEnv("PRISM_ESCROW"),
});

const session = await agent.authenticate();
console.log("authenticated as", session.subject);

const offers = await agent.offers();
if (offers.length === 0) {
  console.log("no GPUs online right now — try again shortly.");
  process.exit(0);
}
console.log(`${offers.length} offer(s) online:`, offers.map((offer) => offer.gpu.model).join(", "));

if (process.env.PRISM_RUN_LEASE !== "1") {
  console.log("\nauth OK. Set PRISM_RUN_LEASE=1 to lease + run (spends USDG + gas).");
  process.exit(0);
}

console.log("leasing a GPU (provisioning takes a few minutes)...");
const lease = await agent.lease({ image: DEFAULT_IMAGE, durationSeconds: 900, minVramMib: 16000 });
console.log("leased", lease.leaseId, "at", `${lease.access.ssh_host}:${lease.access.ssh_port}`);

const result = await agent.run(lease, "nvidia-smi --query-gpu=name,memory.total --format=csv,noheader");
console.log(`\nremote output (exit ${result.code}):\n${result.stdout || result.stderr}`);

agent.endLease(lease);
console.log("lease released. Settlement and a public receipt follow on chain.");

function requireEnv(name) {
  const value = process.env[name];
  if (!value) {
    console.error(`missing ${name} — see the header of this file for the required environment.`);
    process.exit(1);
  }
  return value;
}

File: python-sdk/examples/quickstart.py
"""Authenticate a wallet, list GPUs, and optionally lease one and run a command.

    PRISM_AGENT_KEY=0x<agent wallet private key> \\
    PRISM_ESCROW=0xfD4228eEEfC49e4b76A0CD40af9fdd546220B2FD \\
    python quickstart.py

By default this authenticates and lists GPUs without spending. Set PRISM_RUN_LEASE=1
to lease, run nvidia-smi, and release the lease (spends USDG + gas). Prism is
pre-production and unaudited; do not use funds you cannot lose.
"""
import os
import sys

from prismnetwork import DEFAULT_IMAGE, PrismAgent


def require(name: str) -> str:
    value = os.environ.get(name)
    if not value:
        sys.exit(f"missing {name}")
    return value


agent = PrismAgent(private_key=require("PRISM_AGENT_KEY"), escrow=require("PRISM_ESCROW"))

session = agent.authenticate()
print("authenticated as", session["subject"])

offers = agent.offers()
if not offers:
    sys.exit("no GPUs online right now; try again shortly.")
print(f"{len(offers)} offer(s):", ", ".join(o["gpu"]["model"] for o in offers))

if os.environ.get("PRISM_RUN_LEASE") != "1":
    print("\nauth OK. Set PRISM_RUN_LEASE=1 to lease + run (spends USDG + gas).")
    sys.exit(0)

print("leasing a GPU (provisioning takes a few minutes)...")
lease = agent.lease(image=DEFAULT_IMAGE, duration_seconds=600, min_vram_mib=16000)
print("leased", lease.lease_id, "funded on-chain:", lease.funding_hash)

result = agent.run(lease, "nvidia-smi --query-gpu=name,memory.total --format=csv,noheader")
print(f"\nremote output (exit {result['code']}):\n{result['stdout'] or result['stderr']}")

agent.end_lease(lease)
print("lease released. Settlement and a public receipt follow on chain.")
